Sources_v3/WebSocketClient/EventMiddlewares/ChannelReadUpdaterMiddleware.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A middleware which updates a channel's read events as websocket events arrive.
+struct ChannelReadUpdaterMiddleware<ExtraData: ExtraDataTypes>: EventMiddleware {
+    let database: DatabaseContainer
+    
+    func handle(event: Event, completion: @escaping (Event?) -> Void) {
+        defer { completion(event) }
+        
+        if let event = event as? MessageReadEvent<ExtraData> {
+            updateReadEvent(for: event.cid, userId: event.userId, lastReadAt: event.readAt)
+            return
+        }
+        
+        if let event = event as? NotificationMarkReadEvent<ExtraData> {
+            updateReadEvent(for: event.cid, userId: event.userId, lastReadAt: event.readAt)
+            return
+        }
+        
+        if let event = event as? NotificationMarkAllReadEvent<ExtraData> {
+            let readsForUser = ChannelReadDTO.load(userId: event.userId, context: database.viewContext)
```

Comment:
Same here, you should get the `reads` inside `write { }`

Reply:
Fixed in 60eb7b192816dd80a24b5eb9b8ecae1b9062a851